lib/online-shop-stack.ts

```diff
@@ -0,0 +1,288 @@
+import * as cdk from '@aws-cdk/core';
+import * as appsync from '@aws-cdk/aws-appsync';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as lambdaNodejs from '@aws-cdk/aws-lambda-nodejs';
+import * as cognito from '@aws-cdk/aws-cognito';
+import {CfnUserPoolUserToGroupAttachmentProps} from "@aws-cdk/aws-cognito";
+import * as dynamodb from '@aws-cdk/aws-dynamodb';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn_tasks from '@aws-cdk/aws-stepfunctions-tasks';
+import { AWS_CONFIG } from '../resources/config';
+
+const {
+  OWNER_EMAIL: ownerEmail,
+  CUSTOMER_EMAIL: customerEmail,
+  SES_REGION: sesRegion,
+  SES_ACCOUNT_ID: sesAccountId
+} = AWS_CONFIG;
+
+export class OnlineShopStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    cdk.Tags.of(this).add('Owner', ownerEmail);
+
+    const userPool = new cognito.UserPool(this, 'OnlineShopUserPool', {
+    });
+
+    const onlineShopTable = new dynamodb.Table(this, 'OnlineShopTable', {
+      partitionKey: {
+        name: 'PK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      sortKey: {
+        name: 'SK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+    });
+
+    onlineShopTable.addGlobalSecondaryIndex({
+      indexName: 'GSI1',
+      partitionKey: {
+        name: 'GSI1PK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      sortKey: {
+        name: 'GSI1SK',
+        type: dynamodb.AttributeType.STRING,
+      },
+    });
+
+    const lambdaConfig = {
+      memorySize: 1024,
```

Comment:
[nitpick-ish] I don't believe we need this much memory - default would be fine as well